PR3 - Remove -diff flag from development workflow

juanpprieto opened the pull request:
### WHY are these changes introduced?

Related to #3141 (React Router 7.8.x migration)

The diff-based development system is broken when adopting React Router 7.8.x typegen](https://reactrouter.com/explanation/type-safety) in the skeleton. This system was originally designed for thin example overlays on the Remix skeleton template, but is being replaced with standalone examples that are complete, self-contained RR7 7.8.2 applications with typegen. 

Examples typecheck is broken in both local and CI because RR7's typegen doesn't work with our diff logic because it dynamically builds a project and .react-router type folder can't generate types that are on the skeleton routes (only generates for routes on the examples folder). The complexity to make the flag work with typegen is not worth it so this is a great time to remove some legacy code.

### WHAT is this pull request doing?

This PR removes the `--diff` flag and all associated diff-based development functionality from the Hydrogen CLI:

**1. Complete removal of diff system:**
- Deleted `packages/cli/src/lib/template-diff.ts` (266 lines removed)
- Removed `--diff` flag from all CLI commands (build, dev, preview, codegen, deploy)
- Cleaned up all diff-related imports and logic
- Updated CLI manifest

**2. Temporarily disabled examples/recipes in CI:**
- Removed examples and cookbook from workspace array in `package.json`
- Prevents CI failures since examples are still diff-based (not yet standalone)
- Added placeholder scripts that echo migration status

**3. Added React Router version checking:**
- New `react-router-version-check.ts` utility warns about version mismatches
- Integrated into `dev` command to help users upgrade to React Router 7.8.2
- Provides specific npm commands to fix version issues

**4. Fixed tests for post-diff world:**
- Updated `remote.test.ts` to expect direct template copying (no diff merging)
- Fixed transpile function to handle missing `app/routes.ts` gracefully
- Temporarily skipped 2 tests that require standalone examples (with TODO comments)

### HOW to test your changes?

1. **Run CLI tests:**
   ```bash
   cd packages/cli
   npm test
   # Expected: 282 passed, 2 skipped (documented)
   ```

2. **Verify TypeScript and linting:**
   ```bash
   npm run typecheck  # Should pass
   cd ../..
   npm run lint       # Only unrelated warnings
   ```

3. **Test version checking:**
   ```bash
   cd templates/skeleton
   npm run dev
   # If React Router versions mismatch, you'll see a warning
   ```

4. **Verify examples are isolated from CI:**
   ```bash
   npm run build:examples:disabled
   # Should output: "Examples build temporarily disabled..."
   ```

#### Post-merge steps

None required for this PR.

#### Checklist

- [x] I've read the [Contributing Guidelines](https://github.com/Shopify/hydrogen/blob/main/CONTRIBUTING.md)
- [x] I've considered possible cross-platform impacts (Mac, Linux, Windows)
- [ ] I've added a [changeset](https://github.com/Shopify/hydrogen/blob/main/CONTRIBUTING.md#changesets) if this PR contains user-facing or noteworthy changes
- [x] I've added [tests](https://github.com/Shopify/hydrogen/blob/main/CONTRIBUTING.md#testing) to cover my changes
- [x] I've added or updated the documentation

juanpprieto committed:
[2025-07] Fix Money component compatibility with Customer Account API USDC currency (#3090)

* Fix Money component compatibility with Customer Account API USDC currency

The 2025-07 API update introduced USDC currency code to Customer Account API but not Storefront API, causing TypeScript errors and runtime failures.

Changes:
- Update Money component to accept MoneyV2 from both Storefront and Customer Account APIs via union types
- Enhance useMoney hook to detect unsupported currency codes (like USDC) and gracefully fall back to decimal formatting
- Add currency code suffix for unsupported currencies (e.g., "100.00 USDC") to maintain clarity
- Default to 2 decimal places for USDC to reinforce its 1:1 USD peg based on industry standards

Technical details:
- Handle Intl.NumberFormat RangeError for cryptocurrency codes not in ISO 4217
- Add comprehensive test coverage for both API types and USDC formatting
- Update TypeScript types to support both CurrencyCode enums

Fixes #3089

* Add changeset for Money component USDC compatibility fix

* Add build script to copy customer-account-api-types to dist

Since our Money component now imports from customer-account-api-types, we need to ensure this file is copied to the dist folder during the build process. Previously, only storefront-api-types was being copied.

- Replace copy-storefront-types with copy-api-types script that copies both API type files
- Update build and dev:demo scripts to use the new copy-api-types script
- This fixes the CI build failure where customer-account-api-types.d.ts was not found in dist

* Trigger CI rebuild

kdaviduik committed:
Suggested money changes for PR #3090 (#3097)

kdaviduik committed:
Add @shopify/hydrogen/oxygen export path

Creates new oxygen export with:
- createRequestHandler: Oxygen-specific wrapper around React Router's handler
  - Request validation (GET/HEAD with body check)
  - URL normalization (double slash handling)
  - Powered-by header injection
- getStorefrontHeaders: Extract Oxygen headers from requests

This replaces the functionality from @shopify/remix-oxygen

kdaviduik committed:
Move stack trace improvements to Vite plugin for Oxygen builds

Adds Error.prototype.toString override to Vite's crossBoundarySetup
so it's automatically injected into Oxygen worker environments.
This replaces the runtime override that was in remix-oxygen.

kdaviduik committed:
Update TypeScript references from remix-oxygen to oxygen-workers-types

Updates /// <reference types> directives in env.d.ts files to use
@shopify/oxygen-workers-types instead of @shopify/remix-oxygen

juanpprieto committed:
refactor(cli): remove diff-based development system

Remove the deprecated --diff flag and associated functionality from all
CLI commands. The diff system is no longer needed with React Router 7.8.x
standalone applications.

Changes:
- Remove --diff flag from build, dev, preview, codegen commands
- Delete template-diff.ts and related diff utilities
- Clean up onboarding and live-reload code
- Update CLI manifest and documentation

This simplifies the CLI and removes complexity that's no longer needed.

juanpprieto committed:
test: fix failing tests after diff removal

- Fix plugin-autolinker test by clearing CI environment variables in beforeEach
  The console.log for auto-linking is only shown when not in CI, so tests
  need to ensure CI and GITHUB_ACTIONS env vars are cleared

- Update remote.test.ts to reflect that examples are now copied directly
  Since diff functionality was removed, examples are copied as-is rather
  than having diffs applied. Updated test expectations accordingly.

- Exclude env.d.ts from JS transpilation test since it's a TypeScript-only file

juanpprieto committed:
feat: pin React Router to exact version 7.8.2 across monorepo

- Pin react-router and react-router-dom to exact 7.8.2
- Pin @react-router/* packages to exact 7.8.2
- Remove version ranges (~, ^) to ensure consistency
- Update all packages, templates, and examples
- Update package-lock.json to lock exact versions

juanpprieto committed:
fix: hoist @react-router/dev and fs-routes to fix CLI tests

- Add @react-router/dev and @react-router/fs-routes to root devDependencies
- Ensures packages are available in root node_modules for test symlinks
- Fixes CLI setup test that creates temporary directories

juanpprieto committed:
fix: update .gitignore and example lint scripts for React Router 7.8.2 compatibility

- Add .react-router to .gitignore (generated by React Router 7.8.2)
- Remove deprecated --ext flag from example lint scripts for ESLint 9 compatibility

juanpprieto committed:
fix: update UIMatch test helper for React Router 7.8.2 compatibility

- React Router 7.8.2 requires loaderData to be non-optional in UIMatch type
- Add loaderData property to fillMatch helper in seo.test.ts

juanpprieto committed:
Lint and format

juanpprieto committed:
Lint and format

juanpprieto committed:
Add changeset

juanpprieto committed:
fix: TypeScript 5.9 compatibility updates

- Upgrade TypeScript to 5.9.2 across all packages
- Fix enum compatibility between Storefront and Customer Account APIs
- Fix Navigation type mocks in hydrogen package tests
- Fix BodyInit type compatibility in cache implementation

juanpprieto committed:
fix: resolve eslint dependency issue in subscriptions example

Add missing dependencies to useMemo hook to satisfy react-hooks/exhaustive-deps rule

juanpprieto committed:
fix: properly fix useMemo dependencies in subscriptions example

Move params initialization inside useMemo to avoid recreating on every render

juanpprieto committed:
fix: resolve TypeScript 5.9 compatibility issues with ESLint and update dependencies

This commit addresses critical TypeScript 5.9 compatibility issues that were blocking
the React Router 7.8.x migration. The changes ensure all CI checks pass with zero errors.

## Key Changes:

### ESLint TypeScript Compatibility
- Removed unused `eslint-plugin-hydrogen` package that was causing TypeScript version conflicts
- Updated `@typescript-eslint/eslint-plugin` and `@typescript-eslint/parser` to v8.42.0
- Pinned all ESLint-related package versions for consistency and reproducibility

### Fixed Lint Errors Across Examples
- GTM example: Added missing `ready` and `subscribe` dependencies to useEffect hooks
- Partytown example: Added missing `location.pathname` dependency, fixed iframe accessibility
- Subscriptions example: Replaced deprecated @ts-ignore with @ts-expect-error directives
- Added missing iframe title attributes for accessibility compliance

### Build Configuration
- Added `dist/` to eslint ignores in skeleton template to prevent linting of built files
- Standardized ESLint configuration across all examples using flat config format
- Created eslint.config.js files for all examples to ensure consistent linting

### Dependencies
- Added `@total-typescript/ts-reset@0.6.1` at monorepo level (pinned version)
- Fixed duplicate import issue in hydrogen/src/seo/seo.ts

## Technical Details:

The root cause was eslint-plugin-hydrogen depending on an older version of
@typescript-eslint packages that don't support TypeScript 5.9. Since the plugin
wasn't actively used in the codebase, removing it was the cleanest solution.

All ESLint packages are now pinned to specific versions to prevent future
compatibility issues during dependency updates.

## Testing:
- ✅ All lint checks pass
- ✅ All typecheck passes
- ✅ No ESLint warnings about TypeScript version compatibility
- ✅ All examples build successfully

juanpprieto committed:
ci: trigger CI workflows

juanpprieto committed:
chore: remove pr-plans and changeset files from migration branch

These files were added during development but should not be part of the final PR:
- Removed .github/pr-plans/ directory (planning documentation)
- Removed .changeset/ files added during development

These are internal development artifacts not needed for the React Router 7.8.x migration.

juanpprieto committed:
feat(hydrogen): add React Router 7.8.x support infrastructure

Implement comprehensive React Router 7.8.x support in @shopify/hydrogen package
matching the hydrogen-react-router-7.8.2-clean branch implementation.

Core Infrastructure:
- Add proxy-based hybrid context in createHydrogenContext for React Router compatibility
- Implement unstable_RouterContextProvider with Hydrogen services
- Enable both context.get() and direct property access patterns

New Files:
- Add context-keys.ts with React Router context keys for Hydrogen services
- Add react-router-preset.ts for React Router configuration
- Add react-router.d.ts for module augmentation

Type System Updates:
- Add HydrogenRouterContextProvider interface to types.d.ts
- Update dev.env.d.ts with React Router type augmentation
- Export context keys and NonceProvider from index.ts

Build Configuration:
- Add react-router-preset build targets in tsup.config.ts
- Add react-router-types and react-router-preset exports in package.json
- Include React Router in Vite SSR no-external list

Documentation and Examples:
- Update customer.auth-handler.example.tsx with React Router imports
- Update virtual routes to use react-router imports

All tests pass and skeleton template builds successfully with these changes.

juanpprieto committed:
feat(hydrogen): export hydrogenContext for React Router context.get() pattern

Export a single hydrogenContext object containing all Hydrogen service context keys.
This enables users to access Hydrogen services using React Router's context.get() pattern
while maintaining compatibility with the existing direct property access pattern.

Changes:
- Add hydrogenContext grouped export with all service context keys
- Remove individual context key exports from package (internal use only)
- Add comprehensive tests verifying both access patterns work
- Add documentation and examples for hydrogenContext usage
- Verify proxy-based context supports both get/set methods and direct access

The proxy implementation in createHydrogenContext ensures both patterns work:
1. Direct access: const {storefront, cart} = context
2. Context.get(): const storefront = context.get(hydrogenContext.storefront)

This provides a cleaner API with better discoverability and type safety.

juanpprieto committed:
Add changeset for hydrogen React Router 7.8.x support

juanpprieto committed:
test(hydrogen): add comprehensive tests for React Router 7.8.x integration

- Add tests for HydrogenRouterContextProvider type augmentations
- Add tests for React Router preset configuration and validation
- Add tests for createHydrogenContext proxy behavior
- Add minimal export tests for critical API surface validation
- Test context.get() and context.set() patterns from React Router
- Verify hydrogenContext export structure and API isolation

juanpprieto committed:
fix: apply linting and formatting

juanpprieto committed:
fix(hydrogen-react): update codegen.ts to use Storefront API types for Customer Account enums

- Configure Customer Account API codegen to reference Storefront API types for LanguageCode and CurrencyCode enums
- Ensures type compatibility between APIs when passing i18n.language from Storefront to Customer Account API
- Regenerate Customer Account API types with proper enum references
- Resolves TypeScript errors in skeleton template account routes

juanpprieto committed:
fix: address PR review feedback

- Remove CI_ISSUES_REPORT.md and PR1_STATUS.md files
- Restore deleted changeset file dry-swans-relate.md
- Improve comment for SellingPlanGroup type assertion

juanpprieto committed:
fix: address PR review feedback

- Remove .claude/commands/shopify-cli-update.md file
- Regenerate GraphQL types to ensure correctness
- Add changeset for enum compatibility fix

juanpprieto committed:
chore: merge feat/typescript-lint-rr-7.8 into feat/hydrogen-react-rr-7.8

- Resolved conflicts in codegen.ts by merging both approaches
- Used typescript-lint version of customer-account-api-types.d.ts
- Added CountryCode to enumValues configuration
- Restored .claude/commands/shopify-cli-update.md file

juanpprieto committed:
docs: improve changeset with comprehensive examples for React Router 7.8.x features

juanpprieto committed:
docs: add createRequestHandler export details to changeset

juanpprieto committed:
ci: temporarily disable examples and recipes from CI for React Router 7.8.x migration

- Remove all example folders from npm workspaces array
- Rename build:examples and typecheck:examples scripts to disabled variants
- Update CI workflow step names to reflect only packages and templates are built
- Examples will fail due to skeleton template React Router 7.8.x changes
- This is temporary until examples are updated to match new skeleton patterns

juanpprieto committed:
feat: add React Router version checking to dev command

juanpprieto committed:
fix: make transpile handle missing app/routes.ts for examples

juanpprieto committed:
test: skip example template tests until examples are standalone

- Skip 'creates basic projects from example templates' test
- Skip 'transpiles projects to JS' test
- Both tests fail because --diff flag was removed but examples are still diff-based
- Will be re-enabled in Branch 4 when examples are converted to standalone
- Added TODO comments explaining the situation

juanpprieto committed:
format

juanpprieto committed:
delete .md

juanpprieto committed:
fix: pin React and React-DOM to 18.3.1 across monorepo

- Pin React to exactly 18.3.1 (no ^ or ~) in all packages
- Pin React-DOM to exactly 18.3.1 where used
- This resolves React version mismatch issues in monorepo
- Fixes create-hydrogen test failures due to multiple React instances

juanpprieto committed:
fix: pin React to 18.3.1 across monorepo to resolve version conflicts

- Pin React and React-DOM to exact version 18.3.1 in all packages
- Add npm overrides to force React 18.3.1 for transitive dependencies
- Fixes React hook errors caused by multiple React versions
- All tests now passing (381 tests across 45 files)

juanpprieto committed:
fix: add missing ast-grep Linux x64 optional dependency for CI

- Add @ast-grep/napi-linux-x64-gnu as optional dependency
- Fixes CLI tests failing in Linux CI environment
- Module was missing causing MODULE_NOT_FOUND errors

juanpprieto committed:
fix: remove incorrectly added React dependencies from CLI package

The CLI package doesn't need React as a runtime dependency. These were incorrectly added during the diff removal refactor.

juanpprieto committed:
fix(ci): disable examples deployment and fix build:all script

- Disable examples deployment workflow during React Router 7.8.x migration
- Fix build:all to use build:examples:disabled instead of build:examples
- Examples are not in workspaces and need separate migration

juanpprieto committed:
fix(hydrogen-react): restore separate enum types for Customer Account API

- Remove enumValues override that forced Customer Account API to use Storefront API types
- Allow Customer Account API to generate its own enum types (including USDC currency)
- Update Money component and useMoney hook to handle union types properly
- Fix test helper to use USDC as example of Customer Account API-specific currency
- This properly supports runtime values like USDC that Customer Account API can return

juanpprieto committed:
chore: remove unrelated shopify-cli-update.md changes

This file was inadvertently modified in commit 054c2b39d and should match the base branch version

juanpprieto committed:
docs: update documentation URL to use 2025-07 API version

Address PR #3142 feedback - update outdated 2025-01 documentation URL

juanpprieto committed:
fix: remove unnecessary comments in createHydrogenContext

- Remove comment explaining HydrogenContextOverloads interface
- Remove commented-out function signature that duplicated the actual implementation

juanpprieto committed:
fix: update UnitPriceMeasurement documentation URL to 2025-07

juanpprieto committed:
Regen docs

juanpprieto committed:
feat: add customerAccount.i18n support for Customer Account API

- Added i18n property to CustomerAccount type with language field
- Made I18nBase.language a union of StorefrontLanguageCode | CustomerLanguageCode
- Added cast in createHydrogenContext with explanatory comment
- Updated test snapshots to reflect new context structure

This enables skeleton templates to use context.customerAccount.i18n.language
for Customer Account API operations while maintaining type safety

juanpprieto committed:
test: update i18n replacers snapshot for React Router 7.8.x context structure

rbshop commented on packages/cli/src/lib/onboarding/remote.ts:
```diff
@@ -65,14 +64,7 @@ export async function setupRemoteTemplate(
 
       const {sourcePath, skeletonPath} = downloaded;
```
`skeletonPath` is now unused

rbshop commented on packages/cli/src/lib/react-router-version-check.ts:
```diff
@@ -0,0 +1,134 @@
+  } catch (error) {
+    // If we can't read package.json, skip the check silently
+    // This could happen if running from a different directory or during CI/CD
+    return;
```
We should probably log out when this happens, as this could be a nightmare to debug

juanpprieto committed:
fix: address PR review comments

- Remove unused skeletonPath variable in remote.ts
- Add debug logging when package.json cannot be read for React Router version check

juanpprieto committed:
fix: resolve test failures by fixing temporary directory creation

- Use unique timestamps for test directory names to avoid EEXIST errors
- Add proper timeouts for long-running init and build tests
- Ensure mkdirSync uses recursive option for test directories

juanpprieto committed:
fix: resolve CI test failures by adding build dependencies

- Add build dependency to test task in turbo.json
- Fix mkdirSync usage in CLI tests (no recursive option)
- Ensure tests have required dependencies built first

juanpprieto committed:
chore: remove test project directories

These were accidentally created during test runs

juanpprieto committed:
fix: add SHOPIFY_HYDROGEN_FLAG_LOCKFILE_CHECK to test CI job

The skeleton template doesn't have a lockfile since it's a template,
not a deployed package. The CI test job needs to skip lockfile checks
when running tests that build the skeleton template.
